src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreCommand.cs

```diff
@@ -118,7 +119,7 @@ private void OnRestorePackages(object sender, EventArgs args)
             {
                 _restoreTask = NuGetUIThreadHelper.JoinableTaskFactory
                     .RunAsync(() => SolutionRestoreWorker.Value.ScheduleRestoreAsync(
-                        SolutionRestoreRequest.ByMenu(),
+                        SolutionRestoreRequest.ByMenu(ExplicitRestoreReason.Menu),
```

Comment:
Can we rename to avoid redundant 'menu' word?

```suggestion
                        SolutionRestoreRequest.ByUserCommand(ExplicitRestoreReason.Menu),
````

Reply:
This is a public API; is it possible to change it? But I will change all previous `ByMenu` references to this new function, I will rename that one to this

Reply:
I think that's a good approach.

Reply:
> This is a public API

It's a public API in an assembly that isn't published to nuget.org, so concern is lesser.